pgmpy/inference/mplp.py

```diff
@@ -0,0 +1,241 @@
+from math import log
+import itertools
+import numpy as np
+from operator import sub
+from operator import add
+
+class MPLPAlg:
```

Comment:
* Rename ```MPLPAlg``` to ```MPLP```.
* I am also not convinced with this architecture. Try to inherit it from the base class ```Inference``` and if it is not possible, lets first discuss the architecture before proceeding.